File: py files/Factorial.py
def fact(n):
    f = 1
    if n > 0:
        for i in range(1, n + 1, 1):
            f = f * i
        print("Factorial of number " + str(n) + " is " + str(f))
    else:
        if n == 0:
            print("Factorial of number " + str(n) + " is " + str(f))
        else:
            print("Factorial of negative numbers is invalid")

# Main
print("Enter the number whose factorial is to be calculated")
n = int(input())
fact(n)

File: c files/Factorial.c
#include <stdio.h>

int fact(int n);

int main() {
    printf("Enter the number whose factorial is to be calculated\n");
    int n;
    scanf("%d", &n);
    fact(n);
    return 0;
}

int fact(int n) {
    int f = 1;
    if (n > 0) {
        for (int i = 1; i <= n; i++) {
            f *= i;
        }
        printf("Factorial of number %d is %d\n", n, f);
    }
    else {
        if (n == 0) {
            printf("Factorial of number %d is %d\n", n, f);
        }
        else {
            printf("Factorial of negative numbers is invalid\n");
        }
    }
    return f;
}
